osu.Game/Rulesets/Judgements/DrawableJudgement.cs

```diff
@@ -109,32 +131,46 @@ protected override void PrepareForUse()
 
             prepareDrawables();
 
-            // not sure if this should remain going forward.
-            skinnableJudgement.ResetAnimation();
+            runAnimation();
+        }
 
-            this.FadeInFromZero(FadeInDuration, Easing.OutQuint);
+        private void runAnimation()
+        {
+            ClearTransforms(true);
+            LifetimeStart = Result.TimeAbsolute;
 
-            switch (Result.Type)
+            using (BeginAbsoluteSequence(Result.TimeAbsolute, true))
             {
-                case HitResult.None:
-                    break;
+                // not sure if this should remain going forward.
+                JudgementBody.ResetAnimation();
 
-                case HitResult.Miss:
-                    ApplyMissAnimations();
-                    break;
+                switch (Result.Type)
+                {
+                    case HitResult.None:
+                        break;
 
-                default:
-                    ApplyHitAnimations();
-                    break;
-            }
+                    case HitResult.Miss:
+                        ApplyMissAnimations();
+                        break;
+
+                    default:
+                        ApplyHitAnimations();
+                        break;
+                }
+
+                if (JudgementBody.Drawable is IAnimatableJudgement animatable)
+                {
+                    var drawableAnimation = (Drawable)animatable;
```

Comment:
I'd propose adding lifetime members to `IDrawable` in framework to avoid this casting, but not sure if worth just for this / in line with intentions for that interface...

Reply:
yeah, was disappointed it wasn't there. @smoogipoo any reason these properties don't exists in the interface already?

Reply:
I'd be fine with that, but I don't think it blocks this PR from merge.